include/LightGBM/tree.h

```diff
@@ -501,20 +580,65 @@ inline void Tree::Split(int leaf, int feature, int real_feature,
 }
 
 inline double Tree::Predict(const double* feature_values) const {
-  if (num_leaves_ > 1) {
-    int leaf = GetLeaf(feature_values);
-    return LeafOutput(leaf);
+  if (is_linear_) {
+      int leaf = GetLeaf(feature_values);
+      double output = leaf_const_[leaf];
+      bool nan_found = false;
+      for (size_t i = 0; i < leaf_features_[leaf].size(); ++i) {
+        int feat_raw = leaf_features_[leaf][i];
+        double feat_val = feature_values[feat_raw];
+        if (std::isnan(feat_val)) {
```

Comment:
is that okay to convert all `nan` to zeros, to avoid the constant conversion? which solution is better?

Reply:
I don't think we can change `nan` to zero, this would result in quite different linear models if there are lots of `nan` values.

Reply:
I think that discard nan in linear function, is the same as adding zeros.
assuming x2 have lots of nans. 
`a1x1 + a3x3` is the same as `a1x1 + a2 0 + a3x3`

Reply:
If a feature contains nan, then I am not applying the linear function at all for that row, instead I predict leaf_value (same as standard LightGBM model). Also when I train the linear model, I exclude rows where any feature of the linear model is nan.